executor/seqtest/seq_executor_test.go

```diff
@@ -1312,6 +1312,52 @@ func (s *seqTestSuite) TestPessimisticConflictRetryAutoID(c *C) {
 	}
 }
 
+func (s *seqTestSuite) TestInsertFromSelectConflictRetryAutoID(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t (id int not null auto_increment unique key, idx int unique key, c int);")
+	tk.MustExec("create table src (a int);")
+	concurrency := 2
+	var wg sync.WaitGroup
+	var err []error
+	wgCount := concurrency + 1
+	wg.Add(wgCount)
+	err = make([]error, concurrency)
+	for i := 0; i < concurrency; i++ {
+		tk := testkit.NewTestKitWithInit(c, s.store)
+		go func(idx int) {
+			for i := 0; i < 10; i++ {
+				sql := fmt.Sprintf("insert into t(idx, c) select 1 as idx, 1 as c from src on duplicate key update c = %[1]d", i)
+				_, e := tk.Exec(sql)
+				if e != nil {
+					err[idx] = e
+					wg.Done()
+					return
+				}
+			}
+			wg.Done()
+		}(i)
+	}
+	var insertErr error
+	go func() {
+		tk := testkit.NewTestKitWithInit(c, s.store)
+		for i := 0; i < 10; i++ {
+			_, e := tk.Exec("insert into src values (null);")
+			if e != nil {
+				insertErr = e
+				wg.Done()
+				return
+			}
+		}
+		wg.Done()
+	}()
+	wg.Wait()
+	for _, e := range err {
+		c.Assert(e, IsNil)
```

Comment:
should we use the begin/commit to control the ts of txn2 between the begin and commit of txn1? that will make sure the retry is necessary.

Reply:
It seems the explicit `commit` does not trigger auto-retry:
```sql
mysql> create table t (id int auto_increment unique key, idx int unique key, c int);
Query OK, 0 rows affected (0.00 sec)

mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> insert into t(idx, c) values (1, 1) on duplicate key update c = 2;
Query OK, 1 row affected (0.01 sec)

mysql> commit;
Query OK, 0 rows affected (0.00 sec)
```

```sql
mysql> begin;
Query OK, 0 rows affected (0.00 sec)

mysql> insert into t(idx, c) values (1, 1) on duplicate key update c = 2;
Query OK, 1 row affected (0.00 sec)

mysql> commit;
ERROR 9007 (HY000): Write conflict, txnStartTS=420644278338322432, conflictStartTS=420644277094973440, conflictCommitTS=420644294221365248, key={tableID=49, indexID=2, indexValues={1, }} primary=[]byte(nil) [try again later]
```

Reply:
Because `PessimisticTxn` is set to true by default ...